src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/HandlerProvider.cs

```diff
@@ -50,14 +50,14 @@ public ImmutableArray<RequestHandlerMetadata> GetRegisteredMethods()
         return requestHandlers.Keys.ToImmutableArray();
     }
 
-    private ImmutableDictionary<RequestHandlerMetadata, Lazy<IMethodHandler>> GetRequestHandlers()
+    private ImmutableDictionary<RequestHandlerMetadata, List<Lazy<IMethodHandler, string?>>> GetRequestHandlers()
         => _requestHandlers ??= CreateMethodToHandlerMap(_lspServices);
 
-    private static ImmutableDictionary<RequestHandlerMetadata, Lazy<IMethodHandler>> CreateMethodToHandlerMap(ILspServices lspServices)
+    private static ImmutableDictionary<RequestHandlerMetadata, List<Lazy<IMethodHandler, string?>>> CreateMethodToHandlerMap(ILspServices lspServices)
```

Comment:
Can the language metadata be added to RequestHandlerMetadata instead of adding the separate list?
For normal C# requests, it has no language metadata and maps to the singular right handler
For xaml requests it has language metadata and maps to the singular xaml handler

Reply:
Done